### PR TITLE
source-postgres: re-enable incremental SATs

### LINKED ISSUE #19060: Source Postgres: enable `high` test strictness level in SAT
## What
A `test_strictness_level` field was introduced to Source Acceptance Tests (SAT).
Postgres is a generally_available connector, we want it to have a `high` test strictness level.

**This will help**:
- maximize the SAT coverage on this connector.
- document its potential weaknesses in term of test coverage.

## How
1. Migrate the existing `acceptance-test-config.yml` file to the latest configuration format. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/README.md#L61))
2. Enable `high` test strictness level in `acceptance-test-config.yml`. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/source-acceptance-tests-reference.md#L240))
3. Commit changes on `acceptance-test-config.yml` and open a PR.
4. Run SAT with the `/test` command on the branch.
5. If tests are failing please fix the failing test or use `bypass_reason` fields to explain why a specific test can't be run.

### PR DESCRIPTION
**Pull request by postamar**
Fixes https://github.com/airbytehq/airbyte/issues/19060.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`305f42bc13`](https://github.com/airbytehq/airbyte/pull/31306/commits/305f42bc1338f007d67f643f973ffce4f1e91a21)) - ❌

⏲️  Total pipeline duration: 38mn43s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅        |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                       | ✅        |
| Java Connector Integration Tests                                | ✅        |
| Acceptance tests                                                | ❌        |
| Validate metadata for source-postgres                           | ✅        |
| Connector version semver check                                  | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/postamar_fix-19060/1697050449/305f42bc1338f007d67f643f973ffce4f1e91a21/source-postgres/3.1.9/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:305f42bc1338f007d67f643f973ffce4f1e91a21)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by postamar** [comment]
This is still failing because the test is not picking up the `integration_tests/incremental_configured_catalog.json` catalog for some reason. Will investigate tomorrow.

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`5efb07ec35`](https://github.com/airbytehq/airbyte/pull/31306/commits/5efb07ec35a46f942a42630d16d2486edf7dbac4)) - ✅

⏲️  Total pipeline duration: 20mn14s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ✅       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/postamar_fix-19060/1697659651/5efb07ec35a46f942a42630d16d2486edf7dbac4/source-postgres/3.2.13/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5efb07ec35a46f942a42630d16d2486edf7dbac4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`37d937d1b6`](https://github.com/airbytehq/airbyte/pull/31306/commits/37d937d1b631c20872252d68b2c1d2798196b50f)) - ✅

⏲️  Total pipeline duration: 19mn08s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Build connector tar                                             | ✅       |
| Java Connector Unit Tests                                       | ✅       |
| Build source-postgres docker image for platform(s) linux/x86_64 | ✅       |
| Acceptance tests                                                | ✅       |
| Java Connector Integration Tests                                | ✅       |
| Validate metadata for source-postgres                           | ✅       |
| Connector version semver check                                  | ✅       |
| QA checks                                                       | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/postamar_fix-19060/1697700775/37d937d1b631c20872252d68b2c1d2798196b50f/source-postgres/3.2.13/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:37d937d1b631c20872252d68b2c1d2798196b50f)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Review by bazarnov — APPROVED** [review]

**Review by erohmensing — APPROVED** [review]
Nice one! Looks like the incremental catalog used namespacing (presumably to separate the incremental results from the full refresh ones) which wasnt correctly supported by CAT?

File: airbyte-integrations/bases/connector-acceptance-test/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 2.0.2
+Fix bug in `test_two_sequential_reads` incremental test definition.
```
Comment (alafanechere) [inline airbyte-integrations/bases/connector-acceptance-test/CHANGELOG.md:4]:
```suggestion
Make `test_two_sequential_reads` handle stream namespace
```

**Review by alafanechere — APPROVED** [review]
🎉 
Please bump the CAT version in its Dockerfile

**Comment by alafanechere** [comment]
> Nice one! Looks like the incremental catalog used namespacing (presumably to separate the incremental results from the full refresh ones) which wasnt correctly supported by CAT?

Exactly!

**Comment by postamar** [comment]
Thanks, I addressed all comments.

**Pull request merged**